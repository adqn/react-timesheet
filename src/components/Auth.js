import React, { useState } from 'react';

export default function Auth() {
  const [token, getToken] = (localStorage.getItem(`token`) || null)
  function getAuth() {
    
  }

  return (
    <div>
      <button
        text="Login"
        onClick={() => getAuth()}
      >
      login
      </button>
    </div>
  )
}