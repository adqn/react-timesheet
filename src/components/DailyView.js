import React, { useState, useEffect } from 'react';
import styled from 'styled-components';

const Container = styled.div`
  width: 100%;
  // border: 1px solid red;
`

const Row = styled.div`
  display: flex;
  // flex-direction: row;
  flex-wrap: nowrap;
  // align-content: center;
  justify-content: space-between;
  width: 100%;
  padding: 5px;
  margin-top: -1px;
  border: 1px solid lightgrey;
`

const RowHeader = styled(Row)`
  background: #E9E9E9;
`

const FirstItem = styled.div`
  flex: 1;
  // border: 1px solid black;
  // margin-left: -1px;
  padding: 5px;
`

const Item = styled(FirstItem)`
  text-align: right;
  margin-left: auto
`

const ItemDesc = styled(FirstItem)`
  flex-grow: 2;
  margin-left: 50px;
  margin-right: -50px;
  text-align: left;
  // border: 1px solid red;
`

const Entry = ({data}) =>
  data.map(row =>
    <Row key={row.id}>
      {/* <Item>{row.id}</Item> */}
      <FirstItem>{row.user}</FirstItem>
      <FirstItem>{row.project}</FirstItem>
      <ItemDesc>{row.description}</ItemDesc>
      {/* <Item>{row.progress}</Item> */}
      <Item>{row.hours}h {row.minutes}m</Item>
      <Item>{row.date}</Item>
    </Row>
  )

const DailyView = () => {
  const [loading, setLoading] = useState(true)
  const [testData, setTestData] = useState(null) // Cache this on load

  const getTestData = async () => {
    let timesheet_entries
    const test = await fetch('/api/test')
    timesheet_entries = JSON.parse(test._bodyText).daily2
    setTestData(timesheet_entries)
    setLoading(false)
  }

  useEffect(() => {
    getTestData()
  }, [])

  return (
    <Container>
      <RowHeader>Daily overview</RowHeader>
      {loading ? "Loading..." : <Entry data={testData} />}
    </Container>
  )

}

export default DailyView;