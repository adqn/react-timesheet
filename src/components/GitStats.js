import React, { useState, useEffect } from 'react'
import { Octokit } from 'octokit'
import styled from 'styled-components'

const Container = styled.div`
  display: flex;
  flex-direction: column;
  width: 300px;
  height: 200px;
  border: 1px solid black;
`

export const GitStats = () => {
  const [loading, setLoading] = useState(true);
  const [commits, setCommits] = useState([]);
  const token = localStorage.getItem(`token`)
  const octokit = new Octokit({ auth: token });
  const owner = 'adqn',
    repo = 'react-timesheet',
    perPage = 5;
  
  async function fetchData() {
    const CommitsByWeek = await octokit.request('GET /repos/{owner}/{repo}/stats/commit_activity', {
      owner, repo
    });

    setCommits(CommitsByWeek);
    setLoading(false)
    console.log(CommitsByWeek.data)
  }

  useEffect(() => {
    fetchData();
  }, [])

  return (
    <Container>
      
    </Container>
    )
}