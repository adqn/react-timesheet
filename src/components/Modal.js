import React, { useState, useEffect } from 'react'
import styled from 'styled-components'

const ModalOverlay = styled.div`
  visibility: ${props => props.visibility};
  position: fixed;
  z-index: 1;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background: black;
  background-color: rgba(0,0,0,0.4);
`

export default function Modal(props) {
  return <ModalOverlay
    id="modal"
    visibility={props.visibility}>{props.children}</ModalOverlay>
}