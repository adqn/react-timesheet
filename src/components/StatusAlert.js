import React from 'react';
import styled from'styled-components'

const StatusContainer = styled.div`
  z-index: 1;
`

export const StatusAlert = () => {

}