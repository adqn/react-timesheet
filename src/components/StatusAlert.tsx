import React, { useState, useEffect } from 'react';
import styled, { css } from 'styled-components'

export enum StatusType {
  success = "green",
  failure = "red",
  working = "yellow"
}

const StatusContainer = styled.div <{
  background: StatusType;
  visible: boolean;
}>`
  // visibility: hidden;
  position: fixed;
  left: 50%;
  transform: translate(-50%, 0);
  width: fit-content;
  height: 30px;
  top: 50px;
  padding: 5px;
  border-radius: 5px;
  color: white;
  background: ${props => props.background};
  opacity: 0;
  z-index: 2;
  user-select: none;

  ${props => props.visible && css`
    // visibility: visible;
    transition: opacity 0.3s;
    opacity: 1;
  `};
  ${props => !props.visible && css`
    transition: opacity 0.3s;
    opacity: 0;
    // visibility: hidden;
  `}
`;

export const StatusAlert = (props: {visible: boolean, status: StatusType}) => {
  const [visible, setVisible] = useState<boolean>(props.visible);
  const statusMessages = {
    [StatusType.success]: "Action completed successfully",
    [StatusType.failure]: "Action failed",
    [StatusType.working]: "Working..."
  }

  // useEffect(() => {
  //   if (visible) setTimeout(() => setVisible(false), 4000)
  // }, [visible]);

  return (
    <StatusContainer
      visible={visible}
      background={props.status}
    >
      {statusMessages[props.status]}
    </StatusContainer>
  )
}