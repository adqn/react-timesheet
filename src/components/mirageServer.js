import { createServer } from "miragejs";

// Maybe useful:
// https://github.com/miragejs/examples/blob/master/react-typescript/src/mirage/index.ts

export function makeServer() {
  return createServer({
    routes() {
      this.get("/api/test", () => ({
        stuff: [{ thing: "This has stuff" }],
        daily: [
          {
            user: "Test User",
            project: "React timesheet",
            progress: "1%",
            date: "7-7-2021"
          },
          {
            user: "Test User",
            project: "React timesheet",
            progress: "1.4%",
            date: "7-8-2021"
          },
          {
            user: "Test User",
            project: "React timesheet",
            progress: "1.6%",
            date: "7-9-2021"
          }
        ]
      }))
    },
  });
}