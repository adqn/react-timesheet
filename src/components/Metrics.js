import React, { useState, useEffect } from 'react'
import { Octokit } from 'octokit';
import Plot from 'react-plotly.js'
import * as d3 from 'd3'
import styled from 'styled-components'
import { GitStats } from './GitStats'

// const token = localStorage.getItem(`token`);
// const octokit = new Octokit({auth: token});

// octokit.rest.users.getAuthenticated().then((response) => {
//   console.log(response);
// });


// scale factor is arbitrary but 10 seems good for h = 300px
// so maybe standard scaling something like h/30 or h/25?
// not quite - need a general formula
const coordNorm = (h, y, scale=false) => {
  if (scale) return h - (y * scale)
  else return h - (y * (h/30))
}

const BarChart = () => {
  const d3plot = React.createElement("d3plot")
  const plotInfo = React.createElement("plotInfo")
  
  function drawChart() {
    const data = [3, 5, 10, 2, 15, 2, 4, 20, 9]
    const w = 600,
      h = 300

    const svg = d3
      .select("d3plot")
      .append("svg")
      .style("border", "1px solid black")
      .attr("width", w)
      .attr("height", h)
    
    
    svg.selectAll("rect")
      .data(data)
      .enter()
      .append("rect")
      .attr("x", (d, i) => i * 70)
      .attr("y", (d, i) => coordNorm(h, d)) // rect height within container (but then why second height??)
      .attr("width", 65)
      .attr("height", (d, i) => d * 10)
      .attr("fill", "green")
    
    svg.selectAll("rect")
      .on("mouseover", function () {
        d3.select(this)
          .style("fill", "orange")
      })
      .on("mouseout", function () {
        d3.select(this)
          .style("fill", "green")
      })

    // svg.selectAll("text")
    //   .data(data)
    //   .enter()
    //   .append("text")
    //   .text(d => d)
    //   .attr("x", (d, i) => i * 70)
    //   .attr("y", (d, i) => h - (10 * d) - 3)
  }

  useEffect(() => {
    drawChart()
  }, [])

  return d3plot
}

const D3test = () => {
  const d3test = React.createElement("d3test")
  const data = [1, 2, 3]
  let expanded = false;

  function drawTest() {
    d3.select("d3test")
      // .data(data)
      .html("<div>What</div>")
      .style("width", "100px")
      .style("border", "1px solid black")
      .style("text-align", "center")
      .style("display", "inline-block")
      .style("opacity", "0")
      .on("mousedown", function () {
        if (!expanded) {
          expanded = !expanded
          d3.select(this)
            .transition()
            .duration(1000)
            .style("width", "500px")
        } else {
          expanded = !expanded
          d3.select(this)
            .transition()
            .duration(1000)
            .style("width", "100px")
        }
      })

    d3.select("d3test")
      .data(["some data", "more data yo"])
      .append("div")
      .style("display", "block")
      .style("margin-left", "2px")
      .text(d => d)

    d3.select("d3test")
      .transition()
      .duration(2000)
      .style("opacity", "1")
  }

  useEffect(() => drawTest())

  return d3test
}

const Metrics = () => {
  useEffect(() => {
  })

  return (
    <div>
      <GitStats />
      <D3test />
      <br />
      <BarChart />
    </div>
  )
}

export default Metrics