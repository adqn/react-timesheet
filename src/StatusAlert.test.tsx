import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { StatusAlert, StatusType } from './components/StatusAlert';

describe('StatusAlert', () => {
  test('renders StatusAlert component', () => {
    const tree = renderer
      .create(
        <StatusAlert
          visible={true}
          status={StatusType.success}
        />
      ).toJSON()
    screen.debug();
    expect(tree).toMatchSnapshot();
  });
});