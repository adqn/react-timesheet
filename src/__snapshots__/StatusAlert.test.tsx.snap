// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusAlert renders StatusAlert component 1`] = `
<div
  className="sc-bdvvtL ivTAPF"
  visibility="visible"
>
  Action completed successfully
</div>
`;
